spring-context-indexer/src/test/java/org/springframework/context/index/processor/PropertiesMarshallerTests.java

```diff
@@ -50,6 +55,24 @@ public void readWrite() throws IOException {
 		assertThat(readMetadata.getItems()).hasSize(2);
 	}
 
+	@Test
+	public void metadataIsWrittenDeterministically() throws IOException {
+		CandidateComponentsMetadata metadata = new CandidateComponentsMetadata();
+		metadata.add(createItem("com.b", "type"));
+		metadata.add(createItem("com.c", "type"));
+		metadata.add(createItem("com.a", "type"));
+
+		ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+		PropertiesMarshaller.write(metadata, outputStream);
+		String contents = new String(outputStream.toByteArray(), StandardCharsets.ISO_8859_1);
+		List<String> entries = Stream.of(contents.split(System.lineSeparator()))
```

Comment:
FYI: you do not need to convert the array to a list. AssertJ works nicely with the array. For example, see:

https://github.com/spring-projects/spring-framework/blob/53597f90e9e7da2739da9b874d16a4b1c5c50c10/spring-core/src/test/java/org/springframework/core/SortedPropertiesTests.java#L133-L145

Reply:
Thanks for the hint - I've addressed that.